libvcx/src/init.rs

```diff
@@ -0,0 +1,63 @@
+use futures::Future;
+use settings;
+use indy;
+use indy::{ErrorCode};
+use utils::libindy::pool::{create_pool_ledger_config, open_pool_ledger};
+use utils::libindy::wallet::{set_wallet_handle, build_wallet_credentials, build_wallet_config};
+use indy_sys::WalletHandle;
+use error::{VcxErrorKind, VcxResult, VcxErrorExt};
+
+pub fn init_core(config: &str) -> VcxResult<()> {
+    info!("init_core >>> config = {}", config);
+    settings::process_config_string(&config, true)?;
+    settings::log_settings();
+    ::utils::threadpool::init();
+    Ok(())
+}
+
+pub fn open_pool(pool_name: &str, path: &str, pool_config: Option<&str>) -> VcxResult<()> {
+    trace!("open_pool >>> pool_name={}, path={}, pool_config={:?}", pool_name, path, pool_config);
+
+    create_pool_ledger_config(&pool_name, &path)
+        .map_err(|err| err.extend("Can not create Pool Ledger Config"))?;
+
+    debug!("open_pool ::: Pool Config Created Successfully");
+
+    open_pool_ledger(&pool_name, pool_config)
+        .map_err(|err| err.extend("Can not open Pool Ledger"))?;
+
+    info!("open_pool ::: Pool Opened Successfully");
+
+    Ok(())
+}
+
+pub fn open_as_main_wallet(wallet_name: &str, wallet_key: &str, key_derivation: &str, wallet_type: Option<&str>, storage_config: Option<&str>, storage_creds: Option<&str>) -> VcxResult<WalletHandle> {
```

Comment:
These arguments occur together so often (e.g. devsetup structs) that maybe we can keep them together in a struct. On the other hand, maybe the upkeep would cost more than it's worth...

Reply:
Yeah it can appear so, however this signature is not that common (I found 4 methods with such signature). It's rather the fact that these methods are called on many places (mostly tests). I think having to build a struct to call those methods might make the calling code actually mode cumbersome.